zerver/tests/test_auth_backends.py

```diff
@@ -1511,7 +1519,7 @@ def social_auth_test(self, account_data_dict: Dict[str, str],
 
         return result
 
-    def generate_saml_response(self, email: str, name: str) -> str:
+    def generate_saml_response(self, email: str, name: str, extra_attributes: Dict[str, List[str]]={}) -> str:
```

Comment:
`extra_attributes` is immutable; let's use Mapping/Sequence for it to indicate that.

(I had thought @andersk had added a semgrep rule for this, but basically we never want to create mutable dictionaries this way in the `def` line of a function, because those end up being singletons even though most programmers would not read them that way)

Reply:
Makes sense, fixed!